src/main/java/nextstep/subway/application/LineService.java

```diff
@@ -0,0 +1,82 @@
+package nextstep.subway.application;
+
+import nextstep.subway.domain.Line;
+import nextstep.subway.domain.LineRepository;
+import nextstep.subway.domain.StationRepository;
+import nextstep.subway.dto.LineRequest;
+import nextstep.subway.dto.LineResponse;
+import nextstep.subway.dto.StationResponse;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Service
+@Transactional(readOnly = true)
+public class LineService {
+
+    private LineRepository lineRepository;
+    private StationRepository stationRepository;
+    private static final String NO_SUCH_LINE_EXCEPTION = "해당 ID의 노선 정보가 없습니다.";
+    private static final String NO_SUCH_STATION_EXCEPTION = "해당 ID의 역 정보가 없습니다.";
+
+    public LineService(LineRepository lineRepository, StationRepository stationRepository) {
+        this.lineRepository = lineRepository;
+        this.stationRepository = stationRepository;
+    }
+
+    @Transactional
+    public LineResponse saveLine(LineRequest lineRequest) {
+        Line persistLine = lineRepository.save(lineRequest.toLine());
+        LineResponse lineResponse = getLineResponseWithStations(persistLine);
+        return lineResponse;
+    }
+
+    public List<LineResponse> findAllLines() {
+        List<Line> lines = lineRepository.findAll();
+
+        return lines.stream()
+                .map(line -> getLineResponseWithStations(line))
+                .collect(Collectors.toList());
+    }
+
+    public LineResponse findLineById(Long id) {
+        Line line = lineRepository
+                .findById(id)
+                .orElseThrow(() -> new IllegalArgumentException(NO_SUCH_LINE_EXCEPTION));
+        return getLineResponseWithStations(line);
+    }
+
+    @Transactional
+    public void deleteLineById(Long id) {
+        lineRepository
+                .findById(id)
+                .orElseThrow(() -> new IllegalArgumentException(NO_SUCH_LINE_EXCEPTION));
+        lineRepository.deleteById(id);
+    }
```

Comment:
```suggestion
    @Transactional
    public void deleteLineById(Long id) {
        Line line = lineRepository
                .findById(id)
                .orElseThrow(() -> new IllegalArgumentException(NO_SUCH_LINE_EXCEPTION));
        lineRepository.delete(line);
    }
```
deleteById는 내부적으로 findById를 호출하고있어요!
엔티티를 조회했다면 delete를 호출하는건 어떨까요? 😀
```java
@Transactional
@Override
public void deleteById(ID id) {

Assert.notNull(id, ID_MUST_NOT_BE_NULL);

delete(findById(id).orElseThrow(() -> new EmptyResultDataAccessException(
		String.format("No %s entity with id %s exists!", entityInformation.getJavaType(), id), 1)));
}
```